nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/defn/ViewConfig.java

```diff
@@ -481,6 +481,43 @@
 		String cols() default "";
 
 	}
+	
+	/**
+	 * <p><b>Expected Field Structure</b>
+	 * 
+	 * <p>InputSwitch will be rendered when annotating a field nested under one of the
+	 * following components: <ul><li>{@link Form}</li> <li>{@link Section}</li></ul>
+	 * 
+	 * <p>InputSwitch should decorate a field having a simple type.
+	 * 
+	 * <p>If no orientation is specified, it's considered as DEFAULT. On the need basis
+	 * orientation can be supplied as the LEFT or RIGHT.
+	 * 
+	 * <br>orientation description: <ul><li>DEFAULT orientation places the component right next
+	 * to the Label.</li> <li>LEFT orientation places the component left to the Label.</li> <li>RIGHT
+	 * orientation places the component little away from the Label.</li></ul>
+	 * 
+	 * @since 1.1
+	 */
+	@Retention(RetentionPolicy.RUNTIME)
+	@Target({ElementType.FIELD})
+	@ViewStyle
+	public @interface InputSwitch {
+		public enum Type {
+			LEFT,
+			RIGHT,
+			DEFAULT
+		}
+		String alias() default "InputSwitch";
+		
+		String controlId() default "";
+		
+		String cssClass() default "";
+		
+		boolean postEventOnChange() default false;
+		
+		InputSwitch.Type orientation() default InputSwitch.Type.DEFAULT;
```

Comment:
Please add javadoc for this field as to how it can be used. Specific details from the InputSwitch javadoc can be placed here.